components/analytics-solutions-common/org.wso2.analytics.solutions.common/widgets/DateTimeRangePicker/src/widget/components/TimePicker.jsx

```diff
@@ -0,0 +1,334 @@
+/* eslint-disable react/prop-types,comma-dangle */
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from "react";
+import moment from "moment";
+import Select from "@material-ui/core/Select/Select";
+import MenuItem from "@material-ui/core/MenuItem/MenuItem";
+import Typography from "@material-ui/core/Typography/Typography";
+import TextField from "@material-ui/core/TextField/TextField";
+
+export default class TimePicker extends React.Component {
+  state = {
+    year: this.props.initTime.year(),
+    month: this.props.initTime.month(),
+    days: this.props.initTime.date(),
+    time: this.props.initTime.format("HH:mm:ss.000")
+  };
+
+  /**
+   * Allowing the time step that user can enter according to the input-type
+   * example:If user select hour, user can only change the hour position in the time.
+   * @pram{String} inputType:'hour','minute','second'
+   * @returns{Integer}
+   *
+   */
+  getTimeStep = (inputType) => {
```

Comment:
Instead of a function, this can be defined as an object.

```js
const TimeStep = {
    hour: 3600,
    minute: 60,
    second: 1
}
```

When using,

```js
this.TimeStep[inputType]
```